acertmgr/tools.py

```diff
@@ -257,15 +258,31 @@ def get_cert_valid_until(cert):
 
 
 # @brief convert certificate to PEM format
-# @param cert certificate object in pyopenssl format
+# @param cert certificate object or a list thereof
 # @return the certificate in PEM format
 def convert_cert_to_pem_str(cert):
-    return cert.public_bytes(serialization.Encoding.PEM).decode('utf8')
+    if isinstance(cert, list):
+        result = list()
+        for data in cert:
+            result.append(data.public_bytes(serialization.Encoding.PEM).decode('utf8'))
+        return '\n'.join(result)
+    else:
+        return cert.public_bytes(serialization.Encoding.PEM).decode('utf8')
 
 
 # @brief load a PEM certificate from str
+# @return a certificate object or a list of objects if multiple are in the string
 def convert_pem_str_to_cert(certdata):
-    return x509.load_pem_x509_certificate(certdata.encode('utf8'), default_backend())
+    matches = re.match(r'(-----BEGIN CERTIFICATE-----\n[^\-]+\n-----END CERTIFICATE-----)\n*',
+                       certdata, re.DOTALL)
+    if matches.lastindex < 2:
+        return x509.load_pem_x509_certificate(certdata.encode('utf8'), default_backend())
+    else:
+        result = list()
+        for idx in range(1, matches.lastindex + 1):
+            data = matches.group(idx)
+            result.append(x509.load_pem_x509_certificate(data.encode('utf8'), default_backend()))
```

Comment:
lines 279 and 284 look like a lot of code duplication. What is wrong with always returning a list here?

Reply:
Backwards compatibilty for most of the non-CA code, otherwise a lot more code would have to be updated.